Sources/Identity/CustomerInfo.swift

```diff
@@ -88,6 +88,9 @@ import Foundation
      */
     @objc public let originalApplicationVersion: String?
 
+    /// Dictionary of all subscription product identifiers and their subscription info
+    @objc public let subscriptions: [String: SubscriptionInfo]
```

Comment:
my first thought was that it's a bit confusing that this is a dict but `nonSubscriptions` is an array. I get _why_, but naming-wise I'd maybe have expected both to be the same. Wonder if `subscriptionsByProductIdentifier` would have helped, but not sure. Or maybe even typealiasing `ProductIdentifier = String` so that then you have the type `[ProductIdentifier: SubscriptionInfo]`. 
Then again you could make the argument that we should have all APIs do that or none ¯\\_(ツ)_/¯

Reply:
Instead of a type alias, a [value class](https://kotlinlang.org/docs/inline-classes.html) is also a great candidate for that `ProductIdentifier`. But you'd still have this argument, which I agree with: 
>Then again you could make the argument that we should have all APIs do that or none ¯\(ツ)/¯

Reply:
It's also a bit confusing with `activeSubscriptions` being only the subs ids and this being full objects... Maybe something like `allSubscriptionsInfo`? Or maybe too verbose not sure...

Reply:
Maybe we could deprecate `activeSubscriptions` as part of this change? Not a ton of value in that anymore, and very easy to recreate with `subscriptions`.

Reply:
yes, @fire-at-will made the same comment that maybe `subscriptions` could be a `Set`. Imho I think a dictionary makes more sense since most people using this would want to get a subscription by product identifier, and making it a `Set` forces them to use a filter. Making it a dictionary offers accessing by product id and also offers a `.values` to access only the values.

Thoughts:
- `activeSubscriptions` should really be `activeSubscriptionProductIds` or something like that
- I think I would not want to deprecate `activeSubscriptions`, maybe in the next major, but I still see `activeSubscriptions` valuable since it saves folks from having to do logic on `subscriptions` to get the active ones. I would treat it as syntactic sugar.
- `allSubscriptionsInfo` also makes sense, but I don't think it cleans up the fact that `activeSubscriptions` is just the product ids and `allSubscriptionsInfo` classes. I think it would still missing the informatino that it is a dictionary
- `subscriptionsByProductIdentifier` makes a lot of sense. And the typealias `ProductIdentifier` can also be added. I think that would be the most informative and most correct. I think `byProductIdentifier` is important so we can replicate it in the rest of the SDKs that don't have type aliases